### PR TITLE
Create effect module w/ instances for IO

### PR DESCRIPTION
**Pull request by armanbilge**
So far I've added instances for `Stateful` and `Handle` for `IO`. Any others I can provide? Or for other datatypes in CE3?

### CODE REVIEW

File: tests/src/test/scala/cats/mtl/tests/IOTests.scala [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+
+class IOTests extends BaseSuite with TestInstances {
+
+  implicit val ticker: Ticker = Ticker()
+
+  IOLocal(0)
+    .flatMap { implicit local =>
+      IO.delay(checkAll("Stateful[IO]", StatefulTests[IO, Int].stateful))
+    }
+    .unsafeRunAndForget()
```
Comment (armanbilge) [inline tests/src/test/scala/cats/mtl/tests/IOTests.scala:33]: This test doesn't work right now, and IDK how to make it work. The tricky part is to create an instance of `Stateful[IO, Int]` we need an `IOLocal[Int]` first which only comes in an `IO`.

Comment (djspiewak) [reply tests/src/test/scala/cats/mtl/tests/IOTests.scala:33]: https://github.com/typelevel/munit-cats-effect

Have a look at the `Resource` support! :-)

Comment (armanbilge) [reply tests/src/test/scala/cats/mtl/tests/IOTests.scala:33]: Thanks!! Looking

**Comment by ChristopherDavenport** [comment]
So I think this is a great idea, but for development this does build a cycle

cats-effect depends on cats-mtl for Test scope. 

So whenever a new binary incompatible scala version drops this will need to not be released. Then cats-effect released, and then this module released.

I still think this is probably the right place. But its probably worth noting.

**Comment by armanbilge** [comment]
> So I think this is a great idea, but for development this does build a cycle

Oops. Would this be much easier as a CE module then?

File: tests/src/test/scala/cats/mtl/tests/IOTests.scala (hunk 1 above)
Comment (armanbilge) [reply tests/src/test/scala/cats/mtl/tests/IOTests.scala]: This is now hacked/fixed 🙃

**Comment by armanbilge** [comment]
Via Discord: @djspiewak suggests adding this module would be "dodgy". Alternatives are that this goes in Cats Effect or I spin up a microlib. Also pinging @oleg-py for his thoughts.

Also want to reference #120 which (possibly in some variation) could also find a happy home with effect-related stuff.

Finally, can someone help me figure out the build failure? I don't understand what it's referencing. Thanks!

File: tests/src/test/scala/cats/mtl/tests/IOTests.scala [hunk 2]
```diff
@@ -18,13 +18,14 @@ package cats.mtl.tests
...
 import cats.mtl.effect.instances.io._
-import cats.mtl.laws.discipline.{HandleTests, StatefulTests}
 import cats.effect.IO
 import cats.effect.IOLocal
 import munit.CatsEffectSuite
 import munit.DisciplineSuite
 
 class IOTests extends CatsEffectSuite with DisciplineSuite with TestInstances {
+  import cats.mtl.laws.discipline.{HandleTests, StatefulTests}
+
```
Comment (armanbilge) [inline tests/src/test/scala/cats/mtl/tests/IOTests.scala:28]: I don't understand why the imports need to be here, but the other tests do it too.

**Comment by armanbilge** [comment]
I ended up putting these odds and ends together in a microlib: https://github.com/armanbilge/oxidized

**Pull request closed without merging**